### DISCUSSION TITLE
Any way to resize image under certain size (e.g. 1MB)?

### DISCUSSION THREAD

**Discussion by Zhao-Michael** [SixLabors/ImageSharp#2814] [Q&A]
thanks in advance~

**Comment by JimBobSquarePants** [comment]
In ImageSharp, resizing an image based solely on a target file size (e.g., under 1MB) is not straightforward due to several variables that affect the final file size beyond just the image dimensions. Factors such as:

1. **Image format** (JPEG, PNG, etc.) – Different formats have varying compression algorithms, and some formats (e.g., PNG) are lossless, meaning they will retain more data and result in larger file sizes compared to lossy formats like JPEG.
2. **Quality settings** – Particularly for lossy formats like JPEG or WebP, the compression quality significantly impacts the file size, but adjusting quality to hit an exact size target can be challenging.
3. **Content of the image** – Highly detailed or complex images require more data to store, so two images of the same resolution and quality settings can result in very different file sizes.
4. **Metadata** – Embedded metadata, such as EXIF data, can contribute to file size, and sometimes removing it can help reduce the final size.
5. **Bit depth and color modes** – Images with higher bit depths or color modes (e.g., 32-bit RGBA) take up more space compared to 8-bit or grayscale images.

To achieve a target file size like 1MB, you would need to iteratively resize and adjust the compression quality, and even then, it's not guaranteed to be exact due to the image's content and format.

**Reply by dataogre** [reply]
We are actually implementing an iterative process like this right now. We have an original jpeg, and if the file size isn't small enough to send to a third party, we are compressing it and reducing the dimensions down to a minimum. Once we hit that minimum dimension, we'd like to try grayscale, but I'm struggling to save as grayscale. Does anyone have an example of: given a byte array that is known to be a jpeg, how to convert that jpeg to 8 bit grayscale (which is hopefully has a much smaller file size)? So far, I've failed and haven't been able to find an example anywhere.

**Reply by JimBobSquarePants** [reply]
Use a custom `JpegEncoder` setting the encoding color 

https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Formats.Jpeg.JpegEncodingColor.html
